lib/sequelize.js

```diff
@@ -1095,9 +1095,14 @@ class Sequelize {
       type = dialectTypes[type.key].extend(type);
     }
 
-    if (type instanceof DataTypes.ARRAY && dialectTypes[type.type.key]) {
-      type.type = dialectTypes[type.type.key].extend(type.type);
+    if (type instanceof DataTypes.ARRAY) {
+      if (!type.type) {
+        throw new Error('A value for ARRAY have not been defined.');
```

Comment:
Has. "A value for ARRAY has not been defined." :)

Reply:
Thank you for pointing it out.
I modified it.